### PR TITLE
Merge upstream 5.7.23 into Percona Server

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build links:
https://ps.cd.percona.com/job/percona-server-5.7-param/45/ (OpenSSL)
https://ps.cd.percona.com/job/percona-server-5.7-param/46/ (YaSSL)

The following tests are known to fail in the current Pecona Server 5.7
```
rocksdb.partition
rocksdb.rocksdb_parts
tokudb.bugs.db788-optimize-index-name
tokudb.mvcc-26
tokudb.parts.partition_alter4_tokudb
tokudb.parts.partition_auto_increment_tokudb
tokudb.rows-32m-rand-insert
tokudb.rows-32m-seq-insert
tokudb.rpl.rpl_tokudb_mixed_dml
tokudb.type_datetime
```

```
rocksdb.select
```
This test will be re-recorded because of the new `GROUP BY with ASC/DESC` deprecation warning.

File: client/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ INCLUDE_DIRECTORIES(
   ${CMAKE_SOURCE_DIR}/strings
   ${MY_READLINE_INCLUDE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}
+  SYSTEM ${BOOST_PATCHES_DIR}
```
Comment (laurynas-biveinis) [inline client/CMakeLists.txt]: If my understanding of CMake docs is correct, I don't think it works like this. "SYSTEM" has to come first thing in INCLUDE_DIRECTORIES and will apply to the whole directive, which here has to be split in two: one SYSTEM and one regular

File: libmysqld/CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +32,7 @@ INCLUDE_DIRECTORIES(
   ${SSL_INTERNAL_INCLUDE_DIRS}
   ${CMAKE_SOURCE_DIR}/sql/backup
   ${LZ4_INCLUDE_DIR}
+  SYSTEM ${BOOST_PATCHES_DIR}
```
Comment (laurynas-biveinis) [inline libmysqld/CMakeLists.txt]: Likewise

File: mysql-test/r/percona_ft_query_extra_word_chars.result [hunk 3]
```diff
@@ -47,17 +47,15 @@ INSERT INTO fts_ngram SELECT * FROM words;
 SET SESSION ft_query_extra_word_chars = ON;
 CREATE TABLE fts_ngram_2 LIKE fts_ngram;
 INSERT INTO fts_ngram_2 SELECT * FROM words;
-# Test querying MyISAM in natural language mode, ngram parser, extra chars off
+# Test querying MyISAM in natural language mode, ngram parser, parsed with extra chars off, extra chars off
```
Comment (laurynas-biveinis) [inline mysql-test/r/percona_ft_query_extra_word_chars.result]: Re. that FTS change, let's finish discussing offline first

File: unittest/gunit/keyring_vault/CMakeLists.txt [hunk 4]
```diff
@@ -26,9 +26,9 @@ message(STATUS "Building keyring_vault unittests")
 
 INCLUDE_DIRECTORIES(
         SYSTEM ${GMOCK_INCLUDE_DIRS}
+        SYSTEM ${BOOST_PATCHES_DIR}
```
Comment (laurynas-biveinis) [inline unittest/gunit/keyring_vault/CMakeLists.txt]: Likewise

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: client/CMakeLists.txt (hunk 1 above)
Comment (percona-ysorokin) [reply client/CMakeLists.txt]: Fixed.

File: libmysqld/CMakeLists.txt (hunk 2 above)
Comment (percona-ysorokin) [reply libmysqld/CMakeLists.txt]: Fixed.

File: unittest/gunit/keyring_vault/CMakeLists.txt (hunk 4 above)
Comment (percona-ysorokin) [reply unittest/gunit/keyring_vault/CMakeLists.txt]: Fixed.

**Comment by percona-ysorokin** [comment]
Reverted Oracle's fix for the Bug #25873310
"FULLTEXT SEARCH CAN NOT FIND WORD WHICH CONTAINS ',' OR '.'"
(commit mysql/mysql-server@c1a5784)

Fixed / re-recorded a number of MTR test cases in 'keyring_vault' suite
because of the changed server error messages.

**Comment by percona-ysorokin** [comment]
Updated Jenkins build links
https://ps.cd.percona.com/job/percona-server-5.7-param/53/ (OpenSSL)
https://ps.cd.percona.com/job/percona-server-5.7-param/54/ (YaSSL)

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**